sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/api/yaml/YamlShardingDataSourceFactory.java

```diff
@@ -51,6 +51,27 @@ public static DataSource createDataSource(final File yamlFile) throws SQLExcepti
         return ShardingDataSourceFactory.createDataSource(config.getDataSources(), new ShardingRuleConfigurationYamlSwapper().swap(config.getShardingRule()), config.getProps());
     }
     
+    /**
+     * Create sharding data source.
+     *
+     * @param yamlFile YAML file for rule configuration of databases and tables sharding with data sources
+     * @param configClass target class that the YAML configuration needs to be converted
+     * @return sharding data source
+     * @throws SQLException SQL exception
+     * @throws IOException IO exception
+     */
+    public static DataSource createDataSource(final File yamlFile, Class<? extends YamlRootShardingConfiguration> configClass) throws SQLException, IOException {
```

Comment:
I think this is only for your example code use, in fact yaml is only one of our configuration, we should not change the api easily.